components/layout_2020/positioned.rs

```diff
@@ -553,149 +580,65 @@ impl HoistedAbsolutelyPositionedBox {
-                                .style
-                                .writing_mode
-                                .is_horizontal(),
-                            "Mixed horizontal and vertical writing modes are not supported yet"
-                        );
-
-                        // Clear the context since we will lay out the same descendants
-                        // more than once. Otherwise, absolute descendants will create
-                        // multiple fragments which could later lead to double-borrow
-                        // errors.
-                        positioning_context.clear();
-
-                        let independent_layout = non_replaced.layout(
-                            layout_context,
-                            &mut positioning_context,
-                            &containing_block_for_children,
-                            &containing_block.into(),
-                        );
-
-                        let (block_size, inline_size) =
-                            match independent_layout.content_inline_size_for_table {
-                                Some(inline_size) => {
-                                    (independent_layout.content_block_size, inline_size)
-                                },
-                                None => (
-                                    size.auto_is(|| independent_layout.content_block_size),
-                                    inline_size,
-                                ),
-                            };
-
-                        Result {
-                            content_size: LogicalVec2 {
-                                inline: inline_size,
-                                block: block_size,
-                            },
-                            fragments: independent_layout.fragments,
-                        }
+                    let inline_size = inline_axis.size.non_auto().unwrap();
+                    let containing_block_for_children = ContainingBlock {
+                        inline_size,
+                        block_size: block_axis.size,
+                        style: &style,
+                    };
+                    // https://drafts.csswg.org/css-writing-modes/#orthogonal-flows
+                    assert_eq!(
+                        containing_block.style.writing_mode.is_horizontal(),
+                        containing_block_for_children
+                            .style
+                            .writing_mode
+                            .is_horizontal(),
+                        "Mixed horizontal and vertical writing modes are not supported yet"
+                    );
+
+                    // Clear the context since we will lay out the same descendants
+                    // more than once. Otherwise, absolute descendants will create
+                    // multiple fragments which could later lead to double-borrow
+                    // errors.
+                    positioning_context.clear();
```

Comment:
I guess this is no longer necessary? But I'm not sure about the implications of removing it.

Reply:
If layout only happens once you can and should remove this as it clears all hoisted absolutes from layout.